### ISSUE TITLE
weight_v seems to be the same as the original weight? (About nn.utils.weight_norm())

### ISSUE THREAD

**Issue by xxmy7** [pytorch/pytorch#92809]
### 🐛 Describe the bug

I have one question about the implementation for `nn.utils.weight_norm()`

I read the doc, it says that,

> Weight normalization is a reparameterization that decouples the magnitude of a weight tensor from its direction. This replaces the parameter specified by name (e.g. "weight") with two parameters: one specifying the magnitude (e.g. "weight_g") and one specifying the direction (e.g. "weight_v").

but I find `weight_v` seems to be the same as the original weight.

I try the code as following.

``` python
import torch
import torch.nn as nn

batch_size = 2
time_steps = 3
embedding_dim = 4

inputx = torch.randn(batch_size, time_steps, embedding_dim)  # N * L * C

linear = nn.Linear(embedding_dim, 3, bias=False)
wn_linear = torch.nn.utils.weight_norm(linear)
wn_linear_output = wn_linear(inputx)

weight_direction = linear.weight / linear.weight.norm(dim=1, keepdim=True)

print(linear.weight)
print(weight_direction)
print(wn_linear.weight_v)
```

the output is as follows:
```
tensor([[ 0.2386, -0.1454,  0.4026,  0.2243],
        [-0.3077,  0.3375,  0.1387,  0.2689],
        [-0.2852,  0.2548,  0.1295,  0.1164]],
       grad_fn=<WeightNormInterfaceBackward0>)
tensor([[ 0.4428, -0.2698,  0.7469,  0.4162],
        [-0.5616,  0.6161,  0.2532,  0.4908],
        [-0.6787,  0.6064,  0.3083,  0.2769]], grad_fn=<DivBackward0>)
Parameter containing:
tensor([[ 0.2386, -0.1454,  0.4026,  0.2243],
        [-0.3077,  0.3375,  0.1387,  0.2689],
        [-0.2852,  0.2548,  0.1295,  0.1164]], requires_grad=True)
```

so `weight_v` seems to be the same as the original weight. It's not the same as what I calculated myself(`weight_direction`).

Is there something I'm missing? 

### Versions
```
PyTorch version: 1.13.0
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 专业版
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.13 (main, Oct 13 2022, 21:23:06) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19045-SP0
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3060
Nvidia driver version: 522.25
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.3
[pip3] torch==1.13.0
[pip3] torchaudio==0.13.0
[pip3] torchvision==0.14.0
[conda] blas                      1.0                         mkl
[conda] mkl                       2021.4.0           haa95532_640
[conda] mkl-service               2.4.0            py39h2bbff1b_0
[conda] mkl_fft                   1.3.1            py39h277e83a_0
[conda] mkl_random                1.2.2            py39hf11a4ad_0
[conda] numpy                     1.23.3           py39h3b20f71_0
[conda] numpy-base                1.23.3           py39h4da318b_0
[conda] pytorch                   1.13.0          py3.9_cuda11.7_cudnn8_0    pytorch
[conda] pytorch-cuda              11.7                 h67b0de4_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.13.0                   pypi_0    pypi
[conda] torchvision               0.14.0                   pypi_0    pypi
```

cc @albanD @mruberry @jbschlosser @walterddr @saketh-are

**Comment by albanD** [comment]
Hi,

The direction doesn't have to be 1-norm. To make tracking cheaper, we don't actually normalize the direction all the time but compute the output as `v/||v|| * g` as described in the doc https://pytorch.org/docs/stable/generated/torch.nn.utils.weight_norm.html?highlight=weight_norm#torch.nn.utils.weight_norm

Closing this as expected behavior but feel free to ask follow up if this is not clear!
